v2/emailpassword/troubleshooting/cors-issues.mdx

```diff
@@ -0,0 +1,261 @@
+
+:::danger
+Origin [http://localhost:3000](http://localhost:3000/) is not allowed by Access-Control-Allow-Origin. Status code: 204
+:::
+
+**allowedHeaders [optional] -**
+
+This configures the Access-Control-Allow-Headers CORS header. It expects one of two input types.
+
+- Comma-delimited string - 'X-Custom-Header, Content-Type'
+- Array - ['X-Custom-Header', 'Content-Type']
+
+For Supertokens projects, we can use the SuperTokens.getCORSAllowedHeaders() function which calls the following list - ["anti-csrf", "rid"] for the Content-Type.
+
+This is important because the header tells the browser what HTTP headers are allowed to be used when making a cross-origin request. By default, CORS will reflect the headers specified in the request's Access-Control-Request-Headers header.
+
+**methods [optional] -**
+
+This configures the Access-Control-Allow-Methods CORS header. It expects one of two input types.
+
+- Comma-delimited string - 'GET,PUT,POST'
+- Array - ['GET', 'PUT', 'POST']
+
+This is important because the header tells the browser what HTTP verbs are allowed to be used when making a cross-origin request. Supertokens in particular uses GETs and POSTs to verify, create new accounts, sign in/out and many other functions.
+
+**credentials [must have] -**
+
+This configures the Access-Control-Allow-Credentials CORS header. It expects a boolean.
+
+If you set this header to true, then the browser will allow cookies to be sent in cross-origin requests. This is important because cookies are the only way Supertokens can store the user's session. It is required to set this header to true, so that the browser can send cookies to the Supertokens service.
+
+## Common CORS Issues
+
+:::danger
+[Error] Credentials flag is true, but Access-Control-Allow-Credentials is not "true".
+:::
+
+This happens when the CORS credentials field is not set to true.
+
+We can remedy this issue by setting credentials to ‘true’ in our cors setting. I’ve highlighted the addition in bold below -
+
+```tsx
+import SuperTokens from "supertokens-node";
+import express from "express";
+import cors from "cors";
+let app = express();
+
+// ...other middlewares
+
+app.use(
+  cors({
+    origin: "http://127.0.0.1:8080",
+    allowedHeaders: ["content-type", ...SuperTokens.getAllCORSHeaders()],
+    methods: ["GET", "PUT", "POST", "DELETE"],
+    //highlight-start
+    credentials: true, // Check this line
+    //highlight-end
+
+  })  
+);
+```
+
+:::danger
+[Error] Origin [http://localhost:3000](http://localhost:3000/) is not allowed by Access-Control-Allow-Origin. Status code: 204
+:::
+
+This happens when the CORS origin field is incorrect. Another varient of this error is when the error code is a 404. Generally, that also means a misconfiguration in the CORS origin field. 
```

Comment:
The 404 happens when the OPTIONS API returns a 404 error. This can happen cause of misconfigured API gateway or reverse proxy.

Reply:
Added!